scvid/data/distributed_anndata_dataset.py

```diff
@@ -0,0 +1,28 @@
+import typing as t
+
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from . import DistributedAnnDataCollection
+
+
+class DistributedAnnDataCollectionDataset(Dataset):
+    def __init__(self, dac: DistributedAnnDataCollection) -> None:
+        self.dac = dac
```

Comment:
I'm thinking maybe we can specify here what attributes we want to get in `__getitem__`? One way to do it is by passing a `convert` argument where we can also specify how to convert the data:

```
convert = {
     'X': lambda a: a.toarray() if issparse(a) else a, # densify .X
      'obs': dict(key1 = lambda c: c.astype(dtype)) # change type only for one key of .obs
}

class DistributedAnnDataCollectionDataset(Dataset):
    def __init__(self, dac: DistributedAnnDataCollection, convert: dict) -> None:
        self.dac = dac
        self.dac.convert = convert
        outputs = []
        for name, value in convert.items():
            if isinstance(value, dict):
                for key in value.items():
                    outputs.append(dac.lazy_attr(name, key)
            else:
                outputs.append(dac.lazy_attr(name))
         self.outputs = outputs   # need to order values in the list or have it as a dict
```